com.unity.render-pipelines.universal/Editor/EditorUtils.cs

```diff
@@ -25,46 +26,202 @@ static class EditorUtils
         public const int lwrpAssetCreateMenuPriorityGroup2 = CoreUtils.assetCreateMenuPriority1 + 50;
         public const int lwrpAssetCreateMenuPriorityGroup3 = lwrpAssetCreateMenuPriorityGroup2 + 50;
 
+        internal enum Unit { Metric, Percent }
+
         internal class Styles
         {
             //Measurements
             public static float defaultLineSpace = EditorGUIUtility.singleLineHeight + EditorGUIUtility.standardVerticalSpacing;
-            public static float defaultIndentWidth = 12;
         }
 
-        public static void DrawCascadeSplitGUI<T>(ref SerializedProperty shadowCascadeSplit)
+        public static void DrawCascadeSplitGUI<T>(ref SerializedProperty shadowCascadeSplit, float distance, int cascadeCount, Unit unit)
         {
-            float[] cascadePartitionSizes = null;
-            Type type = typeof(T);
-            if (type == typeof(float))
+            if (cascadeCount <= 0)
             {
-                cascadePartitionSizes = new float[] { shadowCascadeSplit.floatValue };
+                throw new ArgumentException($"Cascade value ({cascadeCount}) needs to be positive.");
             }
-            else if (type == typeof(Vector3))
+
+            int splitCount = cascadeCount - 1;
+
+            float[] cascadePartitionSizes = new float[splitCount];
```

Comment:
Do you need to create the float array here? It seems to be also done in the checks below.